.github/workflows/kubeflow.yml

```diff
@@ -26,33 +38,31 @@ jobs:
           echo "Available storage:"
           df -h
 
-      - uses: actions/checkout@v2
-
-      - name: Set up Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: 3.8
+      - name: Setup environment with Poetry
+        uses: ./.github/actions/setup_environment
 
       - name: Install K3D and kubectl
         run: |
           curl -s https://raw.githubusercontent.com/rancher/k3d/main/install.sh | bash
           curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
           sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+          source $VENV
+          zenml integration install kubeflow -f
 
-      - name: Install poetry
-        run: |
-          python -m pip install --upgrade pip
-          python -m pip install "poetry"
-
-      - name: Install Dependencies
-        run: |
-          python -m poetry install
-          python -m poetry run zenml integration install kubeflow -f
+      - name: Install Linux System Dependencies
+        if: runner.os=='Linux'
+        run: sudo apt install graphviz
 
       - name: Build docker image
         # this step can be much faster if we use an image that copies the
         # lockfile that was generated in the previous step
         run: docker build -t zenml-base-image:latest -f docker/base-dev.Dockerfile .
 
+      - name: Setup tmate session
```

Comment:
This session is only active when the workflow is started manually from the browser and a flag is set manually. It will be helpful for the coming ticket that targets our kubeflow tests